projects/components/src/navigation/navigation-list.component.ts

```diff
@@ -68,20 +73,35 @@ export class NavigationListComponent implements OnChanges {
   @Output()
   public readonly collapsedChange: EventEmitter<boolean> = new EventEmitter();
 
+  @Output()
+  public readonly navItemSelected: EventEmitter<NavItemLinkConfig> = new EventEmitter();
+
   public activeItem$?: Observable<NavItemLinkConfig | undefined>;
 
   public constructor(
     private readonly navigationService: NavigationService,
     private readonly activatedRoute: ActivatedRoute,
-    private readonly navListComponentService: NavigationListComponentService
+    private readonly navListComponentService: NavigationListComponentService,
+    private readonly timeRangeService: TimeRangeService
   ) {}
 
-  public ngOnChanges(): void {
-    this.navItems = this.navListComponentService.resolveFeaturesAndUpdateVisibilityForNavItems(this.navItems);
-    this.activeItem$ = this.navigationService.navigation$.pipe(
-      startWith(this.navigationService.getCurrentActivatedRoute()),
-      map(() => this.findActiveItem(this.navItems))
-    );
+  public ngOnChanges(changes: TypedSimpleChanges<this>): void {
+    if (changes.navItems) {
+      this.navItems = this.navListComponentService.resolveFeaturesAndUpdateVisibilityForNavItems(this.navItems);
+
+      // Initialize the time range service
+      // Depending on FF status, the TR will be either global or page level for the init
+      this.activeItem$ = this.navigationService.navigation$.pipe(
+        startWith(this.navigationService.getCurrentActivatedRoute()),
+        map(() => {
+          const activeItem = this.findActiveItem(this.navItems);
+          if (!this.timeRangeService.isInitialized() && activeItem?.resolveTimeRange) {
+            this.timeRangeService.setDefaultTimeRange(activeItem.resolveTimeRange());
```

Comment:
unclear why we're setting this here

Reply:
I put the logic for calling setDefault in the activeItem$ pipe because it needs the TR value that is decorated on the navItem, and the knowledge of what nav item is currently active, and this captures both, and also this is a shared component.

Reply:
> also this is a shared component.

This is my main concern - it's a shared component, and the whole component is used in contexts where we would not want this behavior - I'm guessing that `resolveTimeRange` is trying to address those cases. I would ideally prefer putting this in the parent left nav instead, unless there's a reason that's not feasible.